api/manifest.proto

```diff
@@ -67,6 +67,16 @@ message Build {
       LINUX_GLIBC = 3;
    }
 
+   // Envoy currently supports [aarch64/arm64, s390x, mips64, ppc, x86_64] as build targets.
+   // See the envoyproxy/envoy repository for more details.
+   // https://github.com/envoyproxy/envoy/blob/main/bazel/BUILD
+   enum Architecture {
+        UNKNOWN_ARCH = 0;
+        X86_64 = 1;
+        ARM64 = 2;
```

Comment:
According to [this letter](https://lists.linaro.org/pipermail/linaro-toolchain/2014-April/004244.html), `aarch64` seems to be appropriate, but Go calls it as `arm64`. I'm not sure which is better.

```
$ go tool dist list | grep linux                                                                                                                 
linux/386
linux/amd64
linux/arm
linux/arm64
linux/mips
linux/mips64
linux/mips64le
linux/mipsle
linux/ppc64
linux/ppc64le
linux/riscv64
linux/s390x
```

Reply:
@musaprg since no one is responding maybe you can find someone who doesn't work here but works on envoy or maybe @mathetake can help. Main thing is arch is relative, and I suspect we want to optimize for the name envoy would use when doing builds, not necessarily what GOARCH says. It is helpful though it they are the same or it doesn't matter.

Reply:
for arm64, I think we should settle on what envoy refers in the docs: arm64
They never say aarch64, yet, so we should be safe

https://www.envoyproxy.io/docs/envoy/latest/search.html?q=arm64&check_keywords=yes&area=default#

counter data point: I noticed the buf project, just now changed their tooling to match linux (aarch64). Any choice we make will have a counter point, but I think knowing who to follow is best. We use upstream conventions first!

https://github.com/bufbuild/buf/releases/tag/v0.42.1
> Change the architecture suffix of the Linux ARM release assets from arm64 to aarch64 to match the output of uname -m on Linux.

Reply:
sheesh problem though, envoy uses the terms depending on OS and also sometimes changes the config name to have a suffix of CPU and other times not.

This is a problem for us when it is missing though we can guess amd64 (aka x86_64) in that case. The other thing we can do is simply accept either. For example, even if they choose aarch when refering to linux and arm when refering to darwin, it isn't the case they choose both for one OS. We can poke for either!

Meanwhile as this problem should bubble up into docker anyway we can also look at how they build their stuff.

https://github.com/envoyproxy/envoy/blob/main/bazel/BUILD

```
selects.config_setting_group(
    name = "apple",
    match_any = [
        ":darwin",
        ":darwin_x86_64",
        ":ios_arm64",
        ":ios_arm64e",
        ":ios_armv7",
        ":ios_armv7s",
        ":ios_i386",
        ":ios_x86_64",
    ],
)

selects.config_setting_group(
    name = "linux",
    match_any = [
        ":linux_aarch64",
        ":linux_mips64",
        ":linux_ppc",
        ":linux_s390x",
        ":linux_x86_64",
    ],
)
```